src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -1,161 +1,132 @@
 using Ocelot.Responses;
 
-namespace Ocelot.DownstreamRouteFinder.UrlMatcher
+namespace Ocelot.DownstreamRouteFinder.UrlMatcher;
+
+public class UrlPathPlaceholderNameAndValueFinder : IPlaceholderNameAndValueFinder
 {
-    public class UrlPathPlaceholderNameAndValueFinder : IPlaceholderNameAndValueFinder
+    private const string PlaceHolderLeft = "{";
+    private const string PlaceHolderRight = "}";
+
+    /// <summary>
+    /// Finds the placeholders in the request path and query and returns their matching values.
+    /// We might encounter the following scenarios:
+    /// - The path template contains a catch-all query parameter. If so, we return the catch-all placeholder with an empty value.
+    /// - The path template contains a catch-all path parameter. If so, we return the catch-all placeholder with an empty value.
+    /// - The path template contains placeholders. We return the placeholders with their matching values.
+    /// </summary>
+    /// <param name="path">The request path.</param>
+    /// <param name="query">The query parameters.</param>
+    /// <param name="pathTemplate">The request path template.</param>
+    /// <returns>The list of the placeholders with their matching values.</returns>
+    public Response<List<PlaceholderNameAndValue>> Find(string path, string query, string pathTemplate)
     {
-        public Response<List<PlaceholderNameAndValue>> Find(string path, string query, string pathTemplate)
+        (bool isCatchAllQuery, string catchAllQueryPlaceholder) = IsCatchAllQuery(pathTemplate);
+        if (isCatchAllQuery)
         {
-            var placeHolderNameAndValues = new List<PlaceholderNameAndValue>();
-
-            path = $"{path}{query}";
-
-            var counterForPath = 0;
-
-            var delimiter = '/';
-            var nextDelimiter = '/';
-
-            for (var counterForTemplate = 0; counterForTemplate < pathTemplate.Length; counterForTemplate++)
+            return new OkResponse<List<PlaceholderNameAndValue>>(new List<PlaceholderNameAndValue>
```

Comment:
Guys, I am tired of the convention of returning `OkResponse<T>` results because I find them overengineered and redundant. If you are refactoring everything, why not simply return a `List<PlaceholderNameAndValue>` object? Please 🙏 Updating the interface is required.

Reply:
Yes, but it has side effects everywhere and it's out of scope

Reply:
Which side effects? Where some tests failed?

Reply:
It doesn't build anymore and I need to update references...